### ISSUE TITLE
Remove cudautils.py

### ISSUE THREAD

**Issue by vyasr** [rapidsai/cudf#10706]
Currently the contents of cudautils.py can be grouped into three sets of functionality:
1. The first set of 7 functions are all targeted at supporting `find_first` and `find_last`, which in turn are used respectively by the `find_first_value` and `find_last_value` methods of Column objects.
2. The second set of functions 4 around window sizes are used by the rolling calculations in rolling.py.
3. The third pair of 2 functions is used for UDF compilation.

The functions in group 3 are necessary but should probably be moved to `core/udf/utils.py`. The remaining functions should be possible to remove altogether, but they may require some additional functionality to be added to libcudf. For instance, `find_first_value` and `find_last_value` may be possible to implement using libcudf's `lower_bound` followed by an equality check, and similarly `find_last_value` could be replaced with `upper_bound`, accounting for all of group 1. For group 2 functions I am not certain if the necessary functionality exists in libcudf's roling aggregations, so we may need to do a little more engineering there first.

**Comment by vyasr** [comment]
@bdice I think you did some recent investigations here, please add any color that I'm missing.

**Comment by bdice** [comment]
I agree with the proposal above. There may be a catch with circular imports for group 3 / UDF helpers.

**Comment by brandon-b-miller** [comment]
I think `find_first` is used in the `iloc` codepath. That part of the code needs to be reworked anyways since it's not producing the right result in some cases.

Let me know if I may be of assistance untangling anything in the UDF area @bdice

**Comment by vyasr** [comment]
> I think `find_first` is used in the `iloc` codepath. That part of the code needs to be reworked anyways since it's not producing the right result in some cases.

Yup, I think that `Column.find_first_value` gets invoked by the `loc_to_iloc` translation. There's also #10657 about trying to unify DataFrame and Series loc/iloc further, so maybe all of the relevant issues could be resolved in one fell swoop.

**Comment by vyasr** [comment]
Assigning to @brandon-b-miller and @wence- for now. I think that Lawrence's work on rolling should make the rolling functions obsolete. After that we can reassess the first set.
